cli-files/go-ls/cmd/root.go

```diff
@@ -0,0 +1,119 @@
+// cmd/root.go
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	ErrInvalidPath           = "Invalid path provided"
+	ErrNotADirectory         = "Provided path is not a directory"
+	ErrCouldNotReadDirectory = "Could not read directory"
+)
+
+var (
+	Delimeter string = "\n"
+	flagM     bool
+)
+
+func Execute() {
+	if err := golsCmd.Execute(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+var golsCmd = &cobra.Command{
+	Use:   "go-ls",
+	Short: "go-ls is an alternative to ls, allows you to list files in a given directory",
+	Long:  "go-ls is an alternative to ls, allows you to list files in a given directory. By default it lists files in the current working directory. You can also specify a directory to list files in.",
+	RunE:  GolsCmdFunc,
+}
+
+func init() {
+	golsCmd.Flags().BoolVarP(&flagM, "m", "m", false, "Use this flag if you would like the output to be comma separated instead of new lines.")
+}
+
+func GolsCmdFunc(cmd *cobra.Command, args []string) error {
+
+	var (
+		path string
+		err  error
+	)
+
+	if flagM {
+		Delimeter = ", "
+	}
+
+	switch len(args) {
+	case 0:
+		path, err = os.Getwd()
+		if err != nil {
+			return err
+		}
+		out, err := ListFolderContents(path)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(cmd.OutOrStdout(), out)
+	case 1:
+		path = args[0]
+		out, err := ListFolderContents(path)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(cmd.OutOrStdout(), out)
+	default:
+		for i, path := range args {
+			out, err := ListFolderContents(path)
+			endOfLine := ""
+			if i < len(args)-1 {
+				endOfLine = "\n"
+			}
+
+			if err != nil {
+				fmt.Println(path + ": " + err.Error() + endOfLine)
+				continue
+			}
+			fmt.Fprint(cmd.OutOrStdout(), path+": \n")
+			fmt.Fprintln(cmd.OutOrStdout(), out+endOfLine)
+		}
+
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ListFolderContents(path string) (string, error) {
+	file, err := os.Stat(path)
+	if err != nil {
+		return "", errors.New(ErrInvalidPath)
```

Comment:
We typically wrap the inner error so that the caller can see the details (right now, this will hid _why_ the path couldn't be statted), with something like `fmt.Errorf("%s: %w", ErrInvalidPath, err)`

Reply:
I have updated this, but not too confident on the new solution. 

I have used `errors.Wrap` to wrap each error, so if the user was to provide a list of paths that cause errors it will wrap all of them together albeit on a new line. 

I fear this is not the correct usage of `Wrap` as it is supposed to give context to 1 error and not accumulate errors? At the same time, cobra doesn't allow returning multiple errors only 1. 

What would be the correct way to approach it?